hls-plugin-api/src/Ide/Plugin/Config.hs

```diff
@@ -73,16 +73,18 @@ instance A.FromJSON Config where
   parseJSON = A.withObject "Config" $ \v -> do
     -- Officially, we use "haskell" as the section name but for
     -- backwards compatibility we also accept "languageServerHaskell"
-    s <- v .: "haskell" <|> v .: "languageServerHaskell"
-    flip (A.withObject "Config.settings") s $ \o -> Config
-      <$> o .:? "hlintOn"                     .!= hlintOn def
-      <*> o .:? "diagnosticsOnChange"         .!= diagnosticsOnChange def
-      <*> o .:? "maxNumberOfProblems"         .!= maxNumberOfProblems def
-      <*> o .:? "diagnosticsDebounceDuration" .!= diagnosticsDebounceDuration def
-      <*> o .:? "liquidOn"                    .!= liquidOn def
-      <*> o .:? "completionSnippetsOn"        .!= completionSnippetsOn def
-      <*> o .:? "formatOnImportOn"            .!= formatOnImportOn def
-      <*> o .:? "formattingProvider"          .!= formattingProvider def
+    c <- v .:? "haskell" <|> v .:? "languageServerHaskell"
+    case c of
+      Nothing -> return def
```

Comment:
I think it should not silently fall back to default configuration. For example, if a user makes a typo and define `hakell` instead, the server won't print any message about it and just hide that it uses fall back config.

Reply:
Would be it be acceptable if we log a message on the server that says something along the line `No user configuration detected. Falling back to default configuration`?

Reply:
Yes, I believe that would give better user experience.

Reply:
I have two solutions in mind for this, none are quite satisfactory for me:
1. Have different constructor for default configuration. We log the message when we determine the config is a default one.
2. Log the config so user can check if their config matches with what the server use.

I changed the PR to solution 1 since it's the most apt solution I can think of. The problem is, we can't use record syntax, at least to my knowledge, with this approach and it looks quite unwieldy.
For solution 2, I think it's not explicit enough for the user to notice.

Let me know if there's a better approach than these two.

Reply:
@aufarg Hi, sorry for the delay, and thanks for the pr.

I think the pr makes sense but it would be nice to keep the record definition.

It would be nice too to print the actual default configuration, keys and values in the log message and show an alert with the short message `No user configuration detected. Falling back to default configuration`